src/runner.rs

```diff
@@ -25,6 +24,59 @@ pub fn runner(requirements: &[Requirements]) -> ExitCode {
     conclusion.exit_code()
 }
 
+fn find_tests(args: &Arguments, requirements: &[Requirements]) -> Vec<Trial> {
+    let tests: Vec<_> = if let Some(exact_filter) = exact_filter(args) {
+        let exact_tests: Vec<_> = requirements
+            .iter()
+            .flat_map(|req| req.exact(exact_filter))
+            .collect();
+
+        if is_nextest() {
+            if exact_tests.is_empty() {
+                panic!("Failed to find exact match for filter {exact_filter}");
+            } else if exact_tests.len() > 1 {
+                panic!(
+                    "Only expected one but found {} exact matches for filter {exact_filter}",
+                    exact_tests.len()
+                );
+            }
+        }
+        exact_tests
+    } else if is_full_scan_forbidden() {
+        panic!("Exact filter was expected to be used");
+    } else {
+        let mut tests: Vec<_> = requirements.iter().flat_map(|req| req.expand()).collect();
+        tests.sort_unstable_by(|a, b| a.name().cmp(b.name()));
+        tests
+    };
+    tests
+}
+
+fn is_nextest() -> bool {
+    std::env::var_os("NEXTEST")
+        .filter(|var| var == "1")
+        .is_some()
+}
+
+fn is_full_scan_forbidden() -> bool {
+    #[cfg(test)]
+    {
+        std::env::var_os("DATATEST_FULL_SCAN_FORBIDDEN")
+            .filter(|var| var == "1")
+            .is_some()
```

Comment:
Oh interesting use of `Option::filter`, presumably to let you compare `OsStr` to `str`? But can this just be `std::env::var`, comparing against `Ok("1")`, since we always want to accept exactly the string `"1"`?

Reply:
Good point.